client/gutenberg/extensions/videopress/edit.js

```diff
@@ -3,55 +3,173 @@
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import Loading from './loading';
+
+const VideoPressEdit = CoreVideoEdit =>
+	class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.state = {
+				media: null,
+				isFetchingMedia: false,
+				fallback: false,
+			};
+		}
 
-const withVideoPressEdit = createHigherOrderComponent( CoreVideoEdit => {
-	return class VideoPressEdit extends Component {
 		componentDidMount() {
-			this.setSrcFromVideoPress();
+			const { guid } = this.props.attributes;
+			if ( ! guid ) {
+				this.setGuid();
+			}
 		}
 
 		componentDidUpdate( prevProps ) {
-			if ( this.props.attributes.id !== prevProps.attributes.id ) {
-				this.setSrcFromVideoPress();
+			const { attributes } = this.props;
+
+			if ( attributes.id !== prevProps.attributes.id ) {
+				this.setGuid();
 			}
 		}
 
-		setSrcFromVideoPress = async () => {
-			const { id, src } = this.props.attributes;
+		fallbackToCore = () => {
+			this.props.setAttributes( { guid: undefined } );
+			this.setState( { fallback: true } );
+		};
+
+		setGuid = async () => {
+			const { attributes, setAttributes } = this.props;
+			const { id } = attributes;
 
-			if ( ! id || isVideoPressUrl( src ) ) {
+			if ( ! id ) {
+				setAttributes( { guid: undefined } );
 				return;
 			}
 
-			const videoPressUrl = await getVideoPressUrl( id );
+			try {
+				this.setState( { isFetchingMedia: true } );
+				const media = await apiFetch( { path: `/wp/v2/media/${ id }` } );
+				this.setState( { isFetchingMedia: false } );
 
-			if ( ! videoPressUrl ) {
-				// No VideoPress data found for given video id
-				return;
-			}
+				const { id: currentId } = this.props.attributes;
+				if ( id !== currentId ) {
+					// Video was changed in the editor while fetching data for the previous video;
+					return;
+				}
 
-			const { id: currentId } = this.props.attributes;
-			if ( id && id !== currentId ) {
-				// Video was changed in the editor while fetching data for the previous video;
-				return;
+				this.setState( { media } );
+				const guid = get( media, 'jetpack_videopress_guid' );
+				if ( guid ) {
+					setAttributes( { guid } );
+				} else {
+					this.fallbackToCore();
+				}
+			} catch ( e ) {
+				this.setState( { isFetchingMedia: false } );
+				this.fallbackToCore();
 			}
+		};
 
+		switchToEditing = () => {
 			this.props.setAttributes( {
-				src: videoPressUrl,
+				id: undefined,
+				guid: undefined,
+				src: undefined,
 			} );
 		};
 
 		render() {
-			return <CoreVideoEdit { ...this.props } />;
+			const {
+				attributes,
+				className,
+				isFetchingPreview,
+				isSelected,
+				isUploading,
+				preview,
+				setAttributes,
+			} = this.props;
+			const { fallback, isFetchingMedia } = this.state;
+
+			if ( isUploading || isFetchingMedia || isFetchingPreview ) {
+				return <Loading text={ isUploading ? __( 'Uploading…' ) : __( 'Embedding…' ) } />;
+			}
```

Comment:
can we keep these separate to avoid confusion?

```js
if ( isUploading ) {
	return <Loading text={ __( 'Uploading…' ) } />;
}

if ( isFetchingMedia || isFetchingPreview ) {
	return <Loading text={ __( 'Embedding…' ) } />
}
```